File: Python/decode_xored_array.py
def decode(encoded: list, first: int) -> list:
    decode_array = [first]
    for i in range(1, len(encoded) + 1):
        temp = decode_array[i - 1] ^ encoded[i - 1]
        decode_array.append(temp)

    return decode_array


if __name__ == "__main__":
    print(decode(encoded = [6,2,7,3], first = 4))

File: core/tmp/Java/src/com/dbc/algorithm/BitOperation/DecodeXoredArray.java
package com.dbc.algorithm.BitOperation;

public class DecodeXoredArray {
    public static int[] decode(int[] encoded, int first) {
        int[] decode = new int[encoded.length + 1];
        decode[0] = first;
        for (int i = 1; i < encoded.length + 1; i++) {
            decode[i] = decode[i - 1] ^ encoded[i - 1];
        }

        return decode;
    }
}
